### DISCUSSION TITLE
Multilevel objects table view

### DISCUSSION THREAD

**Discussion by emilyvanputten** [EvotecIT/PSWriteHTML#181] [Ideas]
I'm looking for a way to visualize objects that have multiple levels of underlying properties in a table like style.
Identical to how you expand a row in "New-HTMLTable", but with the ability to expand 'children of children'.

I've tried different options, explored all examples, but couldn't find what I'm looking for. Maybe it's already contained within the module, but otherwise I hope this could be integrated.

**Comment by PrzemyslawKlys** [comment]
Do you have some visual example showing how do you imagine this? And how you define powershell object for it?

**Reply by emilyvanputten** [reply]
I just made a quick mockup to give an idea. (Don't mind the quality, it could be displayed better ofcourse)

The idea is basically that you can drill down to nested properties. So you're able to present a hierarchical structure in table format (which includes more data richness in comparison with TreeNodes.

![image](https://user-images.githubusercontent.com/25605117/104066269-6baee100-5201-11eb-87bb-b33b5a656532.png)

Another presentation could be to allow the inclusion of additional properties for each node in the treeview format.

**Reply by PrzemyslawKlys** [reply]
I don't think it's going to ever feel  like it should. 

Here's an example of nested table - just a quick one, but it has it's own problems. 

```powershell
Import-Module .\PSWriteHTML.psd1 -Force

New-HTML {
    $Table2 = New-HTMLTable -DataTable (Get-Process | Select-Object -First 1)
    $DataTable3 = @(
        [PSCustomObject] @{
            SomeObject = 'Test2'
            New        = 'Test'
            Table      = [string] $Table2
        }
    )
    New-HTMLTable -DataTable $DataTable3 -InvokeHTMLTags
} -ShowHTML -Online -FilePath $PSScriptRoot\Example7_12.html
```

This blog post shows a way to nest it properly: https://datatables.net/blog/2019-01-11

This comment confirms there's no native way to do row child items: https://datatables.net/forums/discussion/28700/multi-nested-child-rows

The way to go would be to use FancyTree which is implemented (in very basic way) that is able to give pretty advanced output:

![image](https://user-images.githubusercontent.com/15063294/104106198-51820b00-52b4-11eb-8d84-2eb04f7f9636.png)

Here's the demo: https://wwwendt.de/tech/fancytree/demo/#sample-multi-ext.html

I know you said treeview is not enough, but I don't think New-HTMLTable is the way to go.

**Reply by emilyvanputten** [reply]
Thank you for the overview! 
I agree with you that the FancyTree is the way forward to present nested results in a pretty format.
Do you have plans to implement more of its functionality within PSWriteHTML?

**Reply by PrzemyslawKlys** [reply]
It is planned, but as I myself don't have an immediate need to do so it's put into the state - if I will be bored - I may do so. I just have no incentives to work on it. None of my current projects requires this - and you're the only person that wants it.
